src/views/AddItem.jsx

```diff
@@ -1,3 +1,109 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { useState } from 'react';
+import { Button, FormInput } from '../components';
+import { Link } from 'react-router-dom';
+import { addItem } from '../api';
+
+export function AddItem({ listToken }) {
+	const [formFields, setFormFields] = useState({
+		itemName: '',
+		daysUntilNextPurchase: 7,
+	});
+	const [message, setMessage] = useState('');
+
+	const { itemName, daysUntilNextPurchase } = formFields;
+
+	const handleSubmit = async (e) => {
+		e.preventDefault();
+
+		try {
+			await addItem(listToken, {
+				itemName,
+				daysUntilNextPurchase,
+			});
+			setMessage(`${itemName} successfully saved to the database`);
+		} catch (error) {
+			console.log(error);
+			setMessage('item not added to db');
+		}
+	};
+
+	const handleChange = (e) => {
+		if (message) {
+			setMessage('');
+		}
+		setFormFields((prevState) => ({
+			...prevState,
+			[e.target.name]: e.target.value,
+		}));
+	};
+
+	return (
+		<div className="shopping-form-container">
+			{message ? (
+				<div>
+					<h2>{message}</h2>
+					<Link to="/list">
+						<button>View the list</button>
+					</Link>
+				</div>
+			) : (
+				<form onSubmit={handleSubmit}>
+					<FormInput
+						label="Item name"
+						type="text"
+						onChange={handleChange}
+						name="itemName"
+						id="itemName"
+						required
+						placeholder="item name"
+						value={itemName}
+					/>
+					<div className="select-next-purchase-buttons">
+						<fieldset>
+							<legend>How soon will you buy this again?</legend>
+							<div className="radio-btn">
+								<label htmlFor="soon">
+									<input
+										type="radio"
+										name="daysUntilNextPurchase"
+										id="soon"
+										value={7 || daysUntilNextPurchase}
```

Comment:
Why is an OR needed here `7 || daysUntilNextPurchase`? The value is something that doesn't change but always remains the same for each radio button :slightly_smiling_face:

Reply:
@faysvas, it is a fallback to the initial state `daysUntilNextPurchase`.